polynote-spark/src/main/scala/polynote/kernel/interpreter/python/PySparkInterpreter.scala

```diff
@@ -98,11 +98,16 @@ class PySparkInterpreter(
        |# grab the pyspark included in the spark distribution, if available.
        |spark_home = os.environ.get("SPARK_HOME")
        |if spark_home:
-       |    sys.path.insert(1, os.path.join(spark_home, "python"))
+       |    spark_python = os.path.join(spark_home, "python")
+       |    sys.path.insert(1, spark_python)
        |    import glob
        |    py4j_path = glob.glob(os.path.join(spark_home, 'python', 'lib', 'py4j-*.zip'))[0]  # we want to use the py4j distributed with pyspark
        |    sys.path.insert(1, py4j_path)
        |
+       |    pythonpath = os.pathsep.join(filter(lambda x: x is not None, [spark_python, py4j_path, os.environ.get("PYTHONPATH", None)]))
+       |    os.environ["PYTHONPATH"] = pythonpath
+       |    # TODO: we need to set spark.executorEnv.PYTHONPATH somehow (before the session is created)!
```

Comment:
This is a bit of a chicken-and-egg problem: how can we set this before creating the spark session in `LocalSparkKernel` ?

Reply:
Solved by #1006